model/src/gateway/event/gateway.rs

```diff
@@ -97,19 +99,38 @@ impl<'a> GatewayEventDeserializer<'a> {
         input.get(start..start + to)
     }
 
+    // Search for the index of the event type key from the end.
+    fn find_event_type_start(input: &'a str) -> Option<usize> {
+        // If we find it, add 4, since that's the length of what we're searching
+        // for.
+        input.find(r#""t":"#).map(|idx| idx + 4)
+    }
```

Comment:
At a later point it could be nice to have metrics about how often this is called, to see if we should adjust the search from behind value, but it is not needed for the pr.

Reply:
Fixed and made the printed output nicer